include/proxy-wasm/wasm_vm.h

```diff
@@ -311,6 +311,7 @@ class WasmVm {
 
   // Integrator operations.
   std::unique_ptr<WasmVmIntegration> &integration() { return integration_; }
+  virtual void terminateExecution() {}
```

Comment:
1) Could you move it higher? Right now, it's between 2 "integration" functions. Perhaps before `isFailed()`?
2) I think using `terminate` as a name should be enough.
3) Please mark it as a pure virtual function.
4) Some documentation would be useful.